src/renderer/hooks/usePaneNodeSearchMenu.tsx

```diff
@@ -0,0 +1,417 @@
+
+                    if (!sourceType) {
+                        return false;
+                    }
+
+                    const targetTypes = functionDefinitions.get(node.schemaId);
+
+                    if (!targetTypes) {
+                        return false;
+                    }
+
+                    return [...targetTypes.inputs].some(([number, type]) => {
+                        const overlap = intersect(type, sourceType);
+                        return (
+                            overlap.type !== 'never' &&
+                            schemata.get(node.schemaId).inputs[number].hasHandle
+                        );
+                    });
+                }
+                if (connectingFrom.handleType === 'target') {
+                    const sourceFn = typeState.functions.get(connectingFrom.nodeId);
+
+                    if (!sourceFn) {
+                        return false;
+                    }
+
+                    const { inOutId } = parseHandle(connectingFrom.handleId);
+                    const sourceType = sourceFn.inputs.get(inOutId);
+
+                    if (!sourceType) {
+                        return false;
+                    }
+
+                    const targetTypes = functionDefinitions.get(node.schemaId);
+
+                    if (!targetTypes) {
+                        return false;
+                    }
+
+                    return [...targetTypes.outputDefaults].some(([, type]) => {
+                        const overlap = intersect(type, sourceType);
+                        return overlap.type !== 'never';
+                    });
+                }
+                log.error(`Unknown handle type: ${connectingFrom.handleType!}`);
+                return true;
+            }),
+        [connectingFrom, connectingFromType, searchQuery, schemata.schemata]
+    );
+    const byCategories = useMemo(() => getNodesByCategory(matchingNodes), [matchingNodes]);
+
+    const { favorites } = useNodeFavorites();
+
+    useEffect(() => {
+        setSearchQuery('');
+    }, [connectingFrom]);
+
+    const onPaneContextMenuNodeClick = useCallback(
+        (node: NodeSchema, position: Position) => {
+            const reactFlowBounds = wrapperRef.current!.getBoundingClientRect();
+            const { x, y } = position;
+            const projPosition = project({
+                x: x - reactFlowBounds.left,
+                y: y - reactFlowBounds.top,
+            });
+            const nodeId = createUniqueId();
+            const nodeToMake: NodeProto = {
+                id: nodeId,
+                position: projPosition,
+                data: {
+                    schemaId: node.schemaId,
+                },
+                nodeType: node.nodeType,
+            };
+            createNode(nodeToMake);
+            if (isStoppedOnPane && connectingFrom) {
+                const targetTypes = functionDefinitions.get(node.schemaId);
```

Comment:
I would like to get rid of the 4 non-nullish assertions in this code block.
```suggestion
            const targetTypes = functionDefinitions.get(node.schemaId);
            if (isStoppedOnPane && connectingFrom && targetTypes && connectingFromType) {
```